plugins/out_s3/s3.c

```diff
@@ -960,9 +978,17 @@ static int s3_put_object(struct flb_s3 *ctx, const char *tag, time_t create_time
         c = mock_s3_call("TEST_PUT_OBJECT_ERROR", "PutObject");
     }
     else {
-        c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
-                                              uri, body, body_size,
-                                              NULL, 0);
+        if (ctx->canned_acl == NULL) {
+            c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
+                                                  uri, body, body_size,
+                                                  NULL, 0);
+        }
+        else {
+            h = create_canned_acl_header(ctx->canned_acl);
+            c = s3_client->client_vtable->request(s3_client, FLB_HTTP_PUT,
```

Comment:
This plugin has 2 methods of uploading objects:
1. Via the S3 PutObject call which you have modified here
2. Using the multipart upload API, which you have not modified as far as I can tell. I believe that for multipart the canned ACL should be included in the CreateMultipartUpload API request.